internal/provider/data_source_scope.go

```diff
@@ -0,0 +1,101 @@
+package provider
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/hashicorp/boundary/api"
+	"github.com/hashicorp/boundary/api/scopes"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const ()
+
+func dataSourceScope() *schema.Resource {
+	return &schema.Resource{
+		Description: "The scope data source allows you to discover an existing Boundary scope by name.",
+		ReadContext: dataSourceScopeRead,
+
+		Schema: map[string]*schema.Schema{
+			IDKey: {
+				Description: "The ID of the scope.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			NameKey: {
+				Description: "The name of the scope to retrieve.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			DescriptionKey: {
+				Description: "The scope description.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			ScopeIdKey: {
+				Description: "The parent scope ID that will be queried for the scope.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+		},
+	}
+}
+
+func dataSourceScopeRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	md := meta.(*metaData)
+	opts := []scopes.Option{}
+
+	var name string
+	if v, ok := d.GetOk(NameKey); ok {
+		name = v.(string)
+	} else {
+		return diag.Errorf("no name provided")
+	}
+
+	var scopeId string
+	if scopeIdVal, ok := d.GetOk(ScopeIdKey); ok {
+		scopeId = scopeIdVal.(string)
+	} else {
+		return diag.Errorf("no scope ID provided")
+	}
+
+	scp := scopes.NewClient(md.client)
+
+	scpls, err := scp.List(ctx, scopeId, opts...)
+	if err != nil {
+		return diag.Errorf("error calling read scope: %v", err)
+	}
+	if scpls == nil {
+		return diag.Errorf("no scopes found")
+	}
+
+	var scopeIdRead string
+	for _, scopeItem := range scpls.GetItems() {
+		if scopeItem.Name == name {
+			scopeIdRead = scopeItem.Id
+		}
```

Comment:
Nit: once we're found the scope we can break out of the loop early
```suggestion
			break
		}
```

Reply:
Sorry, end of quarter and a new laptop to setup got me a little distracted! I think we are all good now.